with-editor.el

```diff
@@ -662,10 +661,10 @@ else like the former."
 (defun with-editor-shell-command-read-args (prompt &optional async)
   (let ((command (read-shell-command
                   prompt nil nil
-                  (--when-let (or buffer-file-name
-                                  (and (eq major-mode 'dired-mode)
-                                       (dired-get-filename nil t)))
-                    (file-relative-name it)))))
+                  (when-let (file (or buffer-file-name
```

Comment:
@zhenya1007 I am afraid that means that I am not willing to drop the dependency on `dash`, sorry.